packages/patrol/darwin/Classes/AutomatorServer/Automator/IOSAutomator.swift

```diff
@@ -529,34 +529,77 @@
 
     func enableWiFi() throws {
       try runControlCenterAction("enabling wifi") {
-        let toggle = self.springboard.switches["wifi-button"]
-        let exists = toggle.waitForExistence(timeout: self.timeout)
-        guard exists else {
-          throw PatrolError.viewNotExists("wifi-button")
-        }
+        if let osVersionString = self.getOsVersion().split(separator: ".").first,
+          let osVersion = Int(osVersionString)
+        {
+          let toggle: XCUIElement
+
+          if osVersion >= 18 {
+            let wifiOff = self.springboard.images["wifi.slash"].firstMatch
+            let wifiOn = self.springboard.images["wifi"].firstMatch
+            if wifiOn.exists {
+              Logger.shared.i("wifi is already enabled")
+              return
+            }
+            if !wifiOn.exists && !wifiOff.exists {
+              throw PatrolError.viewNotExists("wifi-button")
+            }
+            wifiOn.tap()
+            return
+          } else {
+            toggle = self.springboard.switches["wifi-button"]
+          }
 
-        if toggle.value! as! String == "0" {
-          toggle.tap()
-        } else {
-          Logger.shared.i("wifi is already enabled")
+          let exists = toggle.waitForExistence(timeout: self.timeout)
+          guard exists else {
+            throw PatrolError.viewNotExists("wifi-button")
+          }
+
+          if toggle.value! as! String == "0" {
+            toggle.tap()
+            // Disabling wifi can cause a system alert to appear
+            try self.acceptSystemAlertIfVisible()
+          } else {
+            Logger.shared.i("wifi is already disabled")
+          }
```

Comment:
Should't that code be inside the same block of code as `toggle = self.springboard.switches["wifi-button"]` in line 550? I thought this is the part that gets executed when we are running on iOS < 18